### ISSUE TITLE
"npm install" gives warnings and has vulnerabilities

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-gui#108]
16 high vulnerabilities and several warnings.

```
$ npm install
npm WARN @angular-devkit/build-angular@0.901.0 requires a peer of @angular/compiler-cli@>=9.0.0 < 10 but none is installed. You must install peer dependencies yourself.
npm WARN @angular-devkit/build-angular@0.901.0 requires a peer of typescript@>=3.6 < 3.9 but none is installed. You must install peer dependencies yourself.
npm WARN @ngtools/webpack@9.1.0 requires a peer of @angular/compiler-cli@>=9.0.0 < 10 but none is installed. You must install peer dependencies yourself.
npm WARN @ngtools/webpack@9.1.0 requires a peer of typescript@>=3.6 < 3.9 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of jquery@1.9.1 - 3 but none is installed. You must install peer dependencies yourself.
npm WARN bootstrap@4.3.1 requires a peer of popper.js@^1.14.7 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-moment@3.0.1 requires a peer of @angular/common@>=2.0.0 <7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ngx-moment@3.0.1 requires a peer of @angular/core@>=2.0.0 <7.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN zonemaster-gui@3.1.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.12 (node_modules/webpack-dev-server/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.12: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.12 (node_modules/watchpack/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.12: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/rollup/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.0 (node_modules/karma/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.0: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.12 (node_modules/@angular/compiler-cli/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.12: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

audited 19994 packages in 21.048s

36 packages are looking for funding
  run `npm fund` for details

found 118 vulnerabilities (65 low, 37 moderate, 16 high)
  run `npm audit fix` to fix them, or `npm audit` for details
```

**Comment by matsduf** [comment]
@sandoche2k, have you seen the same issues?

**Comment by pamasse** [comment]
It's really an issue ? Each time you will make npm install, it will check for vulnerabilities in its own database. So, we can do a PR but, tomorrow will have new one ... 
It's more a step in the developpement process than a PR in my point of view.
Each release i update dependencies and i integrate these vulnerabilities patches.

**Comment by matsduf** [comment]
> It's really an issue ? Each time you will make npm install, it will check for vulnerabilities in its own database. So, we can do a PR but, tomorrow will have new one ...

Well, of course it is an issue if we provide code with vulnerabilities. We should try to get rid of as many as possible.

> It's more a step in the developpement process than a PR in my point of view.
> Each release i update dependencies and i integrate these vulnerabilities patches.

I done the "npm install" on Ubuntu 16.04. Is that a good platform to create the test release on? I tested on FreeBSD but it did not work. I would appreciate an instruction what to install on a linux server on which I can successfully run "npm install; npm run release".

When I now test "npm run release" on my Ubuntu, it fails. (See separate issue.)

**Comment by matsduf** [comment]
Resolved by #111.
